### PR TITLE
job-manager: misc cleanup

### PR DESCRIPTION
**Pull request by garlick**
Here are a few job manager/test cleanup commits split off from #4351.    They are  independent of that PR so I thought maybe it's better to submit them separately.

### CODE REVIEW

**Review by chu11 — APPROVED** [review]
LGTM

**Review by grondo — APPROVED** [review]
LGTM, just one question inline.

Also the flux-sched checks failed at
```
 FAIL: t1013-exclusion.t 10 - exclusion: cancel the running job
```
Which I'm not sure how this PR could cause. I'll try restarting the test and see if it was an intermittent test failure.

File: t/t2240-queue-cmd.t [hunk 1]
```diff
@@ -106,8 +106,9 @@ test_expect_success 'flux-queue: status reports reason for stop' '
 	test_cmp status.exp status.out
 '
 
-test_expect_success 'flux-queue: submit 3 jobs' '
-	for i in $(seq 1 3); do flux mini submit /bin/true; done
+test_expect_success 'flux-queue: submit some jobs' '
+	flux mini submit --cc 1-10 /bin/true &&
```
[the remark this answers is not in the document]
Comment (garlick) [reply t/t2240-queue-cmd.t:110]: That would probably be better!  Thanks for that suggestion.  I'll push a fix shortly.

**Comment by grondo** [comment]
> Also the flux-sched checks failed at

Looks like the issue went away on the second run

**Comment by garlick** [comment]
I'm just going to set MWP as I can't see how the coverage report can be correct.

**Pull request merged**